src/shared/ui/common/region.ts

```diff
@@ -0,0 +1,56 @@
+/*!
+ * Copyright 2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import * as nls from 'vscode-nls'
+import { ExtContext } from '../../extensions'
+import { Region } from '../../regions/endpoints'
+import { getRegionsForActiveCredentials } from '../../regions/regionUtilities'
+import { PrompterButtons } from '../buttons'
+import { createQuickPick, QuickPickPrompter } from '../pickerPrompter'
+
+const localize = nls.loadMessageBundle()
+
+type RegionContext = Pick<ExtContext, 'regionProvider' | 'awsContext'>
+type RegionPrompterOptions = {
+    currentRegion?: Region | string
+    title?: string
+    buttons?: PrompterButtons<Region>
+    /** Filter regions based off service availability */
+    serviceId?: string
+}
+
+export function createRegionPrompter(regions: Region[], options: RegionPrompterOptions): QuickPickPrompter<Region>
+export function createRegionPrompter(context: RegionContext, options: RegionPrompterOptions): QuickPickPrompter<Region>
```

Comment:
Not really. I was conflicted over how the regions should be given to this function which is why there are overloads. I'll just make it only accept the global since that'll remove the most duplicate code.